modules/skybrowser/skybrowsermodule.cpp

```diff
@@ -0,0 +1,692 @@
+
+                switch (_interactionMode) {
+                    case MouseInteraction::Hover:
+                        setSelectedObject();
+                        break;
+                    case MouseInteraction::Drag:
+                        _mouseOnPair->translateSelected(_startDragPosition, translation);
+                        break;
+                    case MouseInteraction::FineTune:
+                        _mouseOnPair->fineTuneTarget(_startDragPosition, translation);
+                        break;
+                    default:
+                        setSelectedObject();
+                        break;
+                    }
+                return false;
+            }
+        );
+
+        global::callback::mouseScrollWheel->emplace_back(
+            [&](double, double scroll) -> bool {
+                if (!_isCameraInSolarSystem || !_mouseOnPair || !_allowMouseInteraction) {
+                    return false;
+                }
+                // If mouse is on browser or target, apply zoom
+                _mouseOnPair->setVerticalFovWithScroll(
+                    static_cast<float>(scroll)
+                );
+                return true;
+            }
+        );
+    }
+
+    global::callback::preSync->emplace_back([this]() {
+        // Disable browser and targets when camera is outside of solar system
+        bool camWasInSolarSystem = _isCameraInSolarSystem;
+        glm::dvec3 cameraPos = global::navigationHandler->camera()->positionVec3();
+        _isCameraInSolarSystem = glm::length(cameraPos) < SolarSystemRadius;
+        bool vizModeChanged = _isCameraInSolarSystem != camWasInSolarSystem;
+
+        // Visualization mode changed. Start fading
+        if (vizModeChanged) {
+            _isFading = true;
+            // Camera moved into the solar system
+            if (!_isCameraInSolarSystem) {
+                _goal = Transparency::Transparent;
+            }
+            else {
+                _goal = Transparency::Opaque;
+            }
+        }
+        double deltaTime = global::windowDelegate->deltaTime();
+        // Fade pairs if the camera moved in or out the solar system
+        if (_isFading) {
+            incrementallyFadeBrowserTargets(_goal, deltaTime);
+        }
+        if (_isCameraInSolarSystem) {
+            std::for_each(
+                _targetsBrowsers.begin(), 
+                _targetsBrowsers.end(),
+                [&](const std::unique_ptr<TargetBrowserPair>& pair) {
+                    pair->synchronizeAim();
+                }
+            );
+            incrementallyAnimateTargets(deltaTime);
+        }
+        if (_isCameraRotating && _allowCameraRotation) {
+            incrementallyRotateCamera(deltaTime);
+        }
+    }); 
+} 
+
+void SkyBrowserModule::internalInitialize(const ghoul::Dictionary& dict) {
+    const Parameters p = codegen::bake<Parameters>(dict);
+
+    // Register ScreenSpaceRenderable
```

Comment:
The ""register comments in this function should be removed imo. The code is self-explanatory enough